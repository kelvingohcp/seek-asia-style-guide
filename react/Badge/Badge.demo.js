import React from 'react';
import Badge from './Badge';
import * as sketch from './Badge.sketch';
export default {
  route: '/badge',
  title: 'Badge',
  component: Badge,
  sketch,
  initialProps: {
    label: 'Default'
  },
  options: [
    {
      label: 'Color',
      type: 'radio',
      states: [
        {
          label: 'Default',
          transformProps: props => ({
            ...props,
            label: 'Default',
            color: 'default'
          })
        },
        {
          label: 'Progressing',
          transformProps: props => ({
            ...props,
            label: 'Progressing',
            color: 'progressing'
          })
        },
        {
          label: 'Declined',
          transformProps: props => ({
            ...props,
            label: 'Declined',
            color: 'declined'
          })
        },
        {
          label: 'Expired',
          transformProps: props => ({
            ...props,
            label: 'Expired',
            color: 'expired'
          })
        },
        {
          label: 'New',
          transformProps: props => ({
            ...props,
            label: 'New',
            color: 'new'
          })
        }
      ]
    },
    {
      label: 'is Bold',
      type: 'checklist',
      states: [
        {
          label: 'Bold',
          transformProps: props => ({
            ...props,
            isBold: true
          })
        }
      ]
    }
  ]
};
