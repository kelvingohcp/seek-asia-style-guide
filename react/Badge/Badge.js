import styles from './Badge.less';
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Text from '../Text/Text';

export default function Badge({
  label,
  color,
  isBold
}) {
  let colorStyle = styles.default;
  switch (color) {
    case 'progressing':
      colorStyle = styles.progressing;
      break;
    case 'declined':
      colorStyle = styles.declined;
      break;
    case 'expired':
      colorStyle = styles.expired;
      break;
    case 'new':
      colorStyle = styles.new;
      break;
    case 'default':
    default:
      colorStyle = styles.default;
      break;
  }

  return (
    <span className={classnames({ [styles.root]: true, [styles.isBold]: isBold }, colorStyle)}>
      <Text className={classnames(styles.badgeLabel)} whispering regular={!isBold} strong={isBold}>{label}</Text>
    </span >
  );
}

Badge.propTypes = {
  label: PropTypes.string.isRequired,
  isBold: PropTypes.bool,
  color: PropTypes.oneOf([
    'progressing', 'default', 'declined', 'expired', 'new'
  ])
};

