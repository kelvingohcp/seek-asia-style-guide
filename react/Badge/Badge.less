@import (reference) '~seek-asia-style-guide/theme';

@lighten: #bbb;

.badge {
  border: 1px solid transparent;
  -moz-border-radius: 10px;
  -webkit-border-radius: 10px;
  -o-border-radius: 10px;
  border-radius: 10px;
  box-sizing: border-box;
  color: #333;
  display: inline-block;
  font-size: 12px;
  font-weight: 600;
  line-height: 1;
  padding: 3px 7px;
  white-space: nowrap;
}

.subtle {
  color: @saBlack;
}

.subtleProgressing {
  background-color: screen(@actionGreen, @lighten);
  border-color: screen(@actionGreen, @lighten);
}

.subtleDefault {
  background-color: screen(@actionBlue, @lighten);
  border-color: screen(@actionBlue, @lighten);
}

.subtleDeclined {
  background-color: screen(@actionRed, @lighten);
  border-color: screen(@actionRed, @lighten);
  ;
}

.subtleExpired {
  background-color: screen(@saGrey3, @lighten);
  border-color: screen(@saGrey3, @lighten);
}

.subtleNew {
  background-color: screen(@actionOrange, @lighten);
  border-color: screen(@actionOrange, @lighten);
}

.bold {
  color: @saWhite;
  font-weight: bold;
}

.boldProgressing {
  background-color: @actionGreen;
  border-color: @actionGreen;
}

.boldDefault {
  background-color: @actionBlue;
  border-color: @actionBlue;
}

.boldDeclined {
  background-color: @actionRed;
  border-color: @actionRed;
}

.boldExpired {
  background-color: @saGrey3;
  border-color: @saGrey3;
}

.boldNew {
  background-color: @actionOrange;
  border-color: @actionOrange;
}
