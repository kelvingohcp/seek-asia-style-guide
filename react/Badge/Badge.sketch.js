import React from 'react';
import mapKeys from 'lodash/mapKeys';
import Badge from './Badge';
export const listItem = {
  'Subtle Default': <Badge label="Default" color="default" />,
  'Subtle Progressing': <Badge label="Progressing" color="progressing" />,
  'Subtle Declined': <Badge label="Declined" color="declined" />,
  'Subtle Expired': <Badge label="Expired" color="expired" />,
  'Subtle New': <Badge label="New" color="new" />,
  'Bold Default': <Badge label="Default" color="default" isBold />,
  'Bold Progressing': <Badge label="Progressing" color="progressing" isBold />,
  'Bold Declined': <Badge label="Declined" color="declined" isBold />,
  'Bold Expired': <Badge label="Expired" color="expired" isBold />,
  'Bold New': <Badge label="New" color="new" isBold />
};

// Export text styles as symbols
// (at least until Sketch libraries support shared text styles)
export const symbols = mapKeys(listItem, (value, key) => `Badge/${key}`);
