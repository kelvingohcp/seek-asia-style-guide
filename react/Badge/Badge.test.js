import React from 'react';
import { shallow } from 'enzyme';
import Badge from './Badge';

describe('Badge:', () => {
  it('should render Subtle Default', () => {
    const wrapper = shallow(<Badge label='default' color='default' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Subtle Progressing', () => {
    const wrapper = shallow(<Badge label='Progressing' color='progressing' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Subtle Declined', () => {
    const wrapper = shallow(<Badge label='Declined' color='declined' />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Subtle Expired', () => {
    const wrapper = shallow(<Badge label='Expired' color='expired' />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Subtle New', () => {
    const wrapper = shallow(<Badge label='New' color='new' />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Bold Default', () => {
    const wrapper = shallow(<Badge label='default' color='default' isBold />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Bold Progressing', () => {
    const wrapper = shallow(<Badge label='Progressing' color='progressing' isBold />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Bold Declined', () => {
    const wrapper = shallow(<Badge label='Declined' color='declined' isBold />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Bold Expired', () => {
    const wrapper = shallow(<Badge label='Expired' color='expired' isBold />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Bold New', () => {
    const wrapper = shallow(<Badge label='New' color='new' isBold />);
    expect(wrapper).toMatchSnapshot();
  });
});

