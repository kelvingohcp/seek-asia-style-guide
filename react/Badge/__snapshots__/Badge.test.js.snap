// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge: should render Bold Declined 1`] = `
<span
  className="root isBold declined"
>
  Declined
</span>
`;

exports[`Badge: should render Bold Default 1`] = `
<span
  className="root isBold default"
>
  default
</span>
`;

exports[`Badge: should render Bold Expired 1`] = `
<span
  className="root isBold expired"
>
  Expired
</span>
`;

exports[`Badge: should render Bold New 1`] = `
<span
  className="root isBold new"
>
  New
</span>
`;

exports[`Badge: should render Bold Progressing 1`] = `
<span
  className="root isBold progressing"
>
  Progressing
</span>
`;

exports[`Badge: should render Subtle Declined 1`] = `
<span
  className="root declined"
>
  Declined
</span>
`;

exports[`Badge: should render Subtle Default 1`] = `
<span
  className="root default"
>
  default
</span>
`;

exports[`Badge: should render Subtle Expired 1`] = `
<span
  className="root expired"
>
  Expired
</span>
`;

exports[`Badge: should render Subtle New 1`] = `
<span
  className="root new"
>
  New
</span>
`;

exports[`Badge: should render Subtle Progressing 1`] = `
<span
  className="root progressing"
>
  Progressing
</span>
`;
